runtime/compiler/control/HookedByTheJit.cpp

```diff
@@ -6004,6 +6008,52 @@ static void suspendSamplerThreadForCheckpoint(J9VMThread *samplerThread, J9JITCo
       }
    compInfo->releaseCompMonitor(samplerThread);
    }
+
+static void triggerRecompilationForPreCheckpointGeneratedFSDBodies(J9VMThread *samplerThread, TR_J9VMBase *fe, TR::CompilationInfo *compInfo)
+   {
+   compInfo->acquireCompMonitor(samplerThread);
+
+   J9Method *method;
+   while (method = compInfo->popForcedRecompilation())
+      {
+      J9ROMMethod *romMethod = J9_ROM_METHOD_FROM_RAM_METHOD(method);
+      if (!_J9ROMMETHOD_J9MODIFIER_IS_SET(romMethod, J9AccAbstract)
+            && !_J9ROMMETHOD_J9MODIFIER_IS_SET(romMethod, J9AccNative)
```

Comment:
It could be argued that the methods we filter out (abstract, native and J9AccMethodFrameIteratorSkip) now should not have been put into the list of methods to be recompiled post restore.